connector/wrappers/medchain/client.go

```diff
@@ -0,0 +1,143 @@
+package medchain
+
+import (
+	"sync/atomic"
+
+	"github.com/ldsec/medchain/contracts"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/protobuf"
+	"golang.org/x/xerrors"
+)
+
+// QueryStatus denotes the status of a query.
+type QueryStatus string
+
+const (
+	// QuerySuccessStatus denotes a "success" status
+	QuerySuccessStatus QueryStatus = contracts.QuerySuccessStatus
+	// QueryFailedStatus denotes a "failed" status
+	QueryFailedStatus QueryStatus = contracts.QueryFailedStatus
+)
+
+// Client is used to interact with medchain.
+type Client struct {
+	bcClient *byzcoin.Client
+	signer   darc.Signer
+	counter  uint64
+}
+
+// NewClient creates a new Client by parsing a Byzcoin configuration file
+// at `path` and uses `signer` for signing the transactions
+func NewClient(path string, signer darc.Signer) (*Client, error) {
```

Comment:
yes, to be confirmed by @mickmis . We can also provide the roster file, the skipchain id, and the path to the signer's key.

Reply:
I believe that's OK as long as this is self-contained, i.e. there is no need before calling this function to retrieve stuff etc. 

What's the difference between the roster file and the byzcoin configuration file in that case? From a node, do you need to be aware of the public keys of the others nodes? (I assume yes), so that would be the roster file?

With unlynx we use the roster file to know all the public keys, and a "node index" so that each node is aware of what node it is in the network. This "node index" is available through the configuration.